src/app/public/modules/toggle/toggle.component.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  Component,
+  EventEmitter,
+  forwardRef,
+  Input,
+  Output
+} from '@angular/core';
+
+import {
+  AbstractControl,
+  ControlValueAccessor,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator
+} from '@angular/forms';
+
+import {
+  SkyToggleChange
+} from './types';
+
+/**
+ * Monotonically increasing integer used to auto-generate unique ids for toggle components.
+ */
+let nextId = 0;
+
+/**
+ * Provider Expression that allows sky-toggle to register as a ControlValueAccessor.
+ * This allows it to support [(ngModel)].
+ */
+// tslint:disable:no-forward-ref no-use-before-declare
+export const SKY_TOGGLE_CONTROL_VALUE_ACCESSOR: any = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => SkyToggleComponent),
+  multi: true
+};
+
+const SKY_TOGGLE_VALIDATOR = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => SkyToggleComponent),
+  multi: true
+};
+
+// tslint:enable
+
+@Component({
+  selector: 'sky-toggle',
+  templateUrl: './toggle.component.html',
+  styleUrls: ['./toggle.component.scss'],
+  providers: [
+    SKY_TOGGLE_CONTROL_VALUE_ACCESSOR,
+    SKY_TOGGLE_VALIDATOR
+  ]
+})
+export class SkyToggleComponent implements ControlValueAccessor, Validator {
+
+  /**
+   * Hidden label for screen readers.
+   */
+  @Input()
+  public label: string;
```

Comment:
Gotcha, though it may be worth noting that the other components in `@skyux/forms` with these `aria @Input`s both use `label` and `labelledBy`, rather than the prefixed `ariaLabel` and `ariaLabelledBy`.

Reply:
Ugh, you're correct. We'll have to fix those in a breaking change, soon. Thanks!